examples/hacker_news/hacker_news/repo.py

```diff
@@ -6,39 +6,32 @@
     download_prod_job,
     download_staging_job,
 )
-from .jobs.story_recommender import (
-    story_recommender_local_job,
-    story_recommender_prod_job,
-    story_recommender_staging_job,
-)
+from .jobs.story_recommender import story_recommender
 from .sensors.hn_tables_updated_sensor import make_hn_tables_updated_sensor
 from .sensors.slack_on_failure_sensor import make_slack_on_failure_sensor
+from hacker_news.resources import RESOURCES_LOCAL, RESOURCES_PROD, RESOURCES_STAGING
 
 
 @repository
-def hacker_news_local():
-    return [
-        download_local_job,
-        story_recommender_local_job,
-        dbt_staging_job,
-    ]
-
-
-@repository
-def hacker_news_prod():
-    return [
-        build_schedule_from_partitioned_job(download_prod_job),
-        make_slack_on_failure_sensor(base_url="my_prod_dagit_url.com"),
-        make_hn_tables_updated_sensor(story_recommender_prod_job),
-        make_hn_tables_updated_sensor(dbt_prod_job),
-    ]
-
+def hacker_news(context):
+    if context.deployment_info.deployment_name == "local":
+        return [download_local_job, story_recommender, dbt_staging_job, RESOURCES_LOCAL]
+    elif context.deployment_info.deployment_name == "staging":
+        return [
+            build_schedule_from_partitioned_job(download_staging_job),
+            make_slack_on_failure_sensor(base_url="my_staging_dagit_url.com"),
+            make_hn_tables_updated_sensor(story_recommender),
+            make_hn_tables_updated_sensor(dbt_staging_job),
+            RESOURCES_STAGING,
+        ]
 
-@repository
-def hacker_news_staging():
-    return [
-        build_schedule_from_partitioned_job(download_staging_job),
-        make_slack_on_failure_sensor(base_url="my_staging_dagit_url.com"),
-        make_hn_tables_updated_sensor(story_recommender_staging_job),
-        make_hn_tables_updated_sensor(dbt_staging_job),
-    ]
+    elif context.deployment_nfo.deployment_name == "prod":
+        return [
+            build_schedule_from_partitioned_job(download_prod_job),
+            make_slack_on_failure_sensor(base_url="my_prod_dagit_url.com"),
+            make_hn_tables_updated_sensor(story_recommender),
+            make_hn_tables_updated_sensor(dbt_prod_job),
+            RESOURCES_PROD,
```

Comment:
It feels a little weird to enable including dictionaries in these lists directly. Though I can't think of any particular concrete reason why it's bad.

Maybe: imagine if we continued this pattern and allowed some other kind of dictionary to be included in the repository entities list. And then if someone included an empty dictionary. How would we know if it's an empty resources dictionary vs. an empty other-things dictionary?

Reply:
It wouldn't actually matter right? If it was an empty resources dictionary, and there was an other things dictionary, then we would know. If it was an empty resources dictionary and there was no other things dictionary, then it doesn't actually matter which it is.

Reply:
Yeah, I think you're right. @alangenfeld - curious if you have any reaction to this?

Reply:
fwiw, while I'm not tied to this API, it's kind of refreshing how simple it is IMO.

Reply:
> if you have any reaction to this?

If we end up with an other dict in this list, we could probably disambiguate by asserting all value types in the dicts have to be the same so I don't have a strong argument not to do it.